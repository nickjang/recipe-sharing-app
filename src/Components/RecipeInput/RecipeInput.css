.recipe-input {
  border: 0;
  font-family: var(--note-font);
}

.main-input {
  display: flex;
  flex-direction: column;
}

.recipe-input input {
  border: 1px solid var(--color-black);
}

.recipe-input .input-label {
  margin: auto 4px auto 0px;
}

.recipe-input .hint {
  margin-left: 4px;
}

.recipe-input .validation-error {
  display: block;
  margin: auto;
  max-width: fit-content;
}

@media all and (min-width: 345px) {
  .main-input {
    flex-direction: row;
    justify-content: flex-end;
  }
}
