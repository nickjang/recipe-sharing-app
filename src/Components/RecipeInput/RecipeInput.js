import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ValidationError from '../ValidationError/ValidationError';
import './RecipeInput.css';

class RecipeInput extends Component {
  render() {
    const { form, id, label, value, touched,
      inputRef, ariaLabel, validate, update } = this.props;
    const validationError = (touched &&
      <ValidationError id={id + '-error-message'} errorFor={id} message={validate()} />);

    if (label) {
      return (
        <fieldset className='recipe-input' form={form} name={id}>
          <div className='main-input'>
            {label}
            <input
              type='text'
              id={id}
              value={value}
              name={id}
              ref={inputRef || null}
              aria-describedby={id + '-error-message'}
              aria-invalid={!!validate()}
              onChange={(e) => update(e.target.value)} />
          </div>
          {validationError}
        </fieldset>
      );
    } else if (ariaLabel) {
      return (
        <li className='recipe-input' form={form} name={id}>
          <input
            type='text'
            id={id}
            name={id}
            value={value}
            ref={inputRef || null}
            aria-label={ariaLabel}
            aria-describedby={id + '-error-message'}
            aria-invalid={!!validate()}
            onChange={(e) => update(e.target.value)} />
          {validationError}
        </li>
      );
    }
  }
}

RecipeInput.defaultProps = {
  form: '',
  id: '',
  value: '',
  touched: false,
  validate: () => { },
  update: () => { },
  inputRef: null,
  label: null,
  ariaLabel: ''
}

RecipeInput.propTypes = {
  form: PropTypes.string.isRequired,
  id: PropTypes.string,
  value: PropTypes.string,
  touched: PropTypes.bool,
  validate: PropTypes.func,
  update: PropTypes.func.isRequired,
  inputRef: PropTypes.func,
  label: PropTypes.element,
  ariaLabel: PropTypes.string
}

export default RecipeInput;