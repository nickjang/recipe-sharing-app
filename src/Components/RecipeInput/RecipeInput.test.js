import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RecipeInput from './RecipeInput';

describe(`RecipeInput component`, () => {
  it('renders RecipeInput by default', () => {
    const wrapper = shallow(<RecipeInput />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})