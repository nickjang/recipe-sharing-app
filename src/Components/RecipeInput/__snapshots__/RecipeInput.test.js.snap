// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeInput component renders RecipeInput by default 1`] = `""`;
